### PR TITLE
"validator-added-for-userProjectAdd"

### PR DESCRIPTION
**Pull request by MallanagoudaB**
# Description
These recommendations are intended to promote code quality and team communication during software development. They cover a variety of topics, including ensuring that pull requests are submitted to the correct branch, documenting new methods, preserving consistency across many services, and avoiding typical blunders like accessing APIs or DB queries within loops. Sensitive data should not be uploaded, and changes to environment variables or database models should be executed consistently. Teams may work more effectively and develop higher-quality software by adhering to these standards.

## Type of change
Please choose appropriate options.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Enhancement (additive changes to improve performance)
- [ ]  This change requires a documentation update

## Checklist 

- [ ]  It's critical to avoid making needless file modifications in contributions, such as adding new lines, console logs, or additional spaces, to guarantee cleaner and more efficient code. Furthermore, eliminating unnecessary imports from a file might enhance code readability and efficiency.
- [ ]  Ensure that the pull request is assigned to the right base branch and that the development branch name contains the JIRA Task Id. Furthermore, each commit message should include the JIRA Task Id in the manner "ED-100: message".
- [ ]  Only update packages if it is mentioned and authorized in the design document, and make sure that you have the required permissions.
- [ ]  Avoid making API and database queries inside a loop as it can lead to performance issues and slow down the system.
- [ ]  When calling another function inside a given function, add comments explaining the purpose and meaning of the passed arguments and expected return values.
- [ ]  If adding a blank argument in a function, add a comment explaining the reason for the blank argument.
- [ ]  Before submitting a pull request, do a self-review of your code to ensure there are no conflicts with the base branch and all comments have been addressed.
- [ ]  Before merging a pull request, it's important to have other team members review it to catch any potential errors or issues
- [ ]  To maintain code integrity, it's important to remove all related changes when removing code during a code review.
- [ ]  If new constants, endpoints, or utility functions are introduced, it is important to check if they already exist in the service to avoid any duplication.
- [ ]  Whenever a new environment variable is added to a service, it's important to ensure that the necessary changes are made to related files such as ".env.sample" and "envVariables.js" to maintain consistency and avoid errors. Additionally, the new environment variable should be added to the devops repository to ensure that it is properly documented and accessible to the team.
- [ ]  When adding a new function to a service, it is important to document it with relevant information such as the name, parameters, and return value in a consistent format across all services. Additionally, if there are any changes to the API response, ensure that the documentation in the controllers is updated accordingly. 
- [ ]  Write a clear and concise commit message that describes the changes made.
- [ ]  Maintain consistent function signature and code across all services when adding a function to multiple services. Implement changes to database models in all services that use the same model. 
- [ ]  Use only let and const. Do not use var.
- [ ]  Make common functions for repetitive code blocks.
- [ ]  Avoid uploading sensitive information such as secret tokens or passwords in pull requests to ensure data security.
- [ ]  Maintain consistent indentation and spacing throughout the code.

### CODE REVIEW

File: module/userProjects/helper.js [hunk 1]
```diff
@@ -1630,6 +1630,8 @@ module.exports = class UserProjectsHelper {
 				let projectData = await _projectData(data)
 				if (projectData && projectData.success == true) {
 					createProject = _.merge(createProject, projectData.data)
+				} else {
+					return reject(projectData)
```
Comment (aks30) [inline module/userProjects/helper.js:1634]: @MallanagoudaB why this change required for add API category validation?

File: module/userProjects/validator/v1.js [hunk 2]
```diff
@@ -6,38 +6,36 @@
...
+		tasksStatus: function () {
+			req.checkParams('_id').exists().withMessage('required project id')
+		},
+		solutionDetails: function () {
+			req.checkParams('_id').exists().withMessage('required project id')
+			req.checkQuery('taskId').exists().withMessage('required task id')
+		},
+		add: function () {
+			req.checkBody('title').exists().withMessage('required project title')
+			req.checkBody('categories[0].value').exists().withMessage('required categories[0].value')
```
Comment (aks30) [inline module/userProjects/validator/v1.js:23]: @MallanagoudaB This is not the way to do, what if categories[1].value is empty? Also we should check ids in DB, they cannot give any random ids

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 1 Aug

File: module/userProjects/helper.js (hunk 1 above)
Comment (MallanagoudaB) [reply module/userProjects/helper.js:1634]: Because if the id is not there in DB this will handle it with below message.
{
    "status": 400,
    "message": "Not found category"
}

File: module/userProjects/validator/v1.js (hunk 2 above)
Comment (MallanagoudaB) [reply module/userProjects/validator/v1.js:23]: Okay i have added a validator if the id is not present in DB will throw {
    "status": 400,
    "message": "Not found category"
}
